src/snapred/backend/service/NormalizationService.py

```diff
@@ -147,11 +149,62 @@ def normalization(self, request: NormalizationRequest):
             detectorPeaks=ingredients.detectorPeaks,
         ).dict()
 
+    def validateRequest(self, request: NormalizationRequest):
+        """
+        Validate the normalization request.
+
+        :param request: a normalization request
+        :type request: NormalizationRequest
+        """
+        if not self._sameStates(request.runNumber, request.backgroundRunNumber):
+            raise ValueError("Run number and background run number must be of the same Instrument State.")
+
+        # This is a redundant call, but it is placed here to facilitate re-sequencing.
+        permissionsRequest = CalibrationWritePermissionsRequest(
+            runNumber=request.runNumber, continueFlags=request.continueFlags
+        )
+        self.validateWritePermissions(permissionsRequest)
+
+    def validateWritePermissions(self, request: CalibrationWritePermissionsRequest):
+        """
+        Validate that the normalization-calibration workflow will be able to save its output.
+
+        :param request: a write-permissions request containing the run number and existing continue flags
+        :type request: CalibrationWritePermissionsRequest
+        """
+        # Note: this is split-out as a separate method so it can be checked as early as possible in the workflow.
+        continueFlags = ContinueWarning.Type.UNSET
+
+        # check that the user has write permissions to the save directory
+        if not self.checkWritePermissions(request.runNumber):
+            continueFlags |= ContinueWarning.Type.NO_WRITE_PERMISSIONS | ContinueWarning.Type.CONTINUE_NOT_POSSIBLE
+
+        # remove any continue flags that are present in the request by xor-ing with the flags
+        if request.continueFlags:
+            continueFlags = continueFlags ^ (request.continueFlags & continueFlags)
+
+        if continueFlags:
+            raise ContinueWarning(
+                "<p>It looks like you don't have permissions to write to "
```

Comment:
it would be cool to eventually migrate these templatized messages over to the `application.yml`

Reply:
I think it's standard practice for most larger codebases to have a central "string table".  I wonder about adding it to `application.yml` however.  The problem is that the end users actually see that file.  (And, for SNAPRed, unfortunately it has started functioning like `Mantid.properties`.)  My point here is that there possibly needs to be a separate layer.

Reply:
Oh, which reminds me:  maybe we need a method that gives us (from the _templatized_ string) the _un-templatized_ strings (i.e. without the tags, and with "<br>" replaced by "\n")?